valkey_glide_otel.c

```diff
@@ -0,0 +1,265 @@
+    g_otel_initialized    = true;
+    VALKEY_LOG_INFO("otel_init", "OpenTelemetry initialized successfully");
+    return 1;
+}
+
+/**
+ * Shutdown OpenTelemetry
+ */
+void valkey_glide_otel_shutdown(void) {
+    if (g_otel_config.enabled) {
+        g_otel_config.enabled = false;
+        g_otel_initialized    = false;
+        VALKEY_LOG_INFO("otel_shutdown", "OpenTelemetry shutdown complete");
+    }
+}
+
+/**
+ * Create a span for command tracing
+ */
+uint64_t valkey_glide_create_span(enum RequestType request_type) {
+    if (!g_otel_config.enabled) {
+        return 0;
+    }
+    return create_otel_span(request_type);
+}
+
+/**
+ * Drop a span
+ */
+void valkey_glide_drop_span(uint64_t span_ptr) {
+    if (span_ptr != 0) {
+        drop_otel_span(span_ptr);
+    }
+}
+
+/**
+ * Parse OTEL configuration from PHP array
+ * Following Java/Go validation patterns
+ */
+int parse_otel_config_array(zval* config_array, valkey_glide_otel_config_t* otel_config) {
+    HashTable* ht = Z_ARRVAL_P(config_array);
+
+    /* Allocate main config */
+    otel_config->config = ecalloc(1, sizeof(struct OpenTelemetryConfig));
+
+    /* Set default flush interval (5000ms like Java/Go) */
+    otel_config->config->has_flush_interval_ms = true;
+    otel_config->config->flush_interval_ms     = 5000;
+
+    /* Parse traces configuration */
+    zval* traces_val = zend_hash_str_find(ht, "traces", sizeof("traces") - 1);
+    if (traces_val && Z_TYPE_P(traces_val) == IS_ARRAY) {
+        otel_config->traces_config = ecalloc(1, sizeof(struct OpenTelemetryTracesConfig));
+        HashTable* traces_ht       = Z_ARRVAL_P(traces_val);
+
+        /* Parse endpoint */
+        zval* endpoint_val = zend_hash_str_find(traces_ht, "endpoint", sizeof("endpoint") - 1);
+        if (endpoint_val && Z_TYPE_P(endpoint_val) == IS_STRING) {
+            otel_config->traces_config->endpoint = estrdup(Z_STRVAL_P(endpoint_val));
+        } else {
+            VALKEY_LOG_ERROR("otel_config",
+                             "Traces endpoint is required when traces config is provided");
+            return 0;
+        }
+
+        /* Parse sample_percentage with default of 1% (like Java/Go) */
+        zval* sample_val =
+            zend_hash_str_find(traces_ht, "sample_percentage", sizeof("sample_percentage") - 1);
+        if (sample_val && Z_TYPE_P(sample_val) == IS_LONG) {
+            long sample_pct = Z_LVAL_P(sample_val);
+            if (sample_pct < 0 || sample_pct > 100) {
+                VALKEY_LOG_ERROR("otel_config", "Sample percentage must be between 0 and 100");
+                return 0;
+            }
+            otel_config->traces_config->has_sample_percentage = true;
+            otel_config->traces_config->sample_percentage     = (uint32_t) sample_pct;
+        } else {
+            /* Default to 1% like Java/Go */
+            otel_config->traces_config->has_sample_percentage = true;
+            otel_config->traces_config->sample_percentage     = 1;
+        }
```

Comment:
Similar comment to above. We should leave defaults to Rust core to avoid duplication and maintainability issues.


```suggestion
        /* Parse sample_percentage */
        zval* sample_val =
            zend_hash_str_find(traces_ht, "sample_percentage", sizeof("sample_percentage") - 1);
        if (sample_val && Z_TYPE_P(sample_val) == IS_LONG) {
            long sample_pct = Z_LVAL_P(sample_val);
            if (sample_pct < 0 || sample_pct > 100) {
                VALKEY_LOG_ERROR("otel_config", "Sample percentage must be between 0 and 100");
                return 0;
            }
            otel_config->traces_config->has_sample_percentage = true;
            otel_config->traces_config->sample_percentage     = (uint32_t) sample_pct;
        }
```